src/ray/core_worker/core_worker.cc

```diff
@@ -155,6 +157,37 @@ CoreWorkerProcess::CoreWorkerProcess(const CoreWorkerOptions &options)
   // NOTE(kfstorm): any initialization depending on RayConfig must happen after this line.
   InitializeSystemConfig();
 
+  // Begin to get gcs server address from raylet.
+  gcs_server_address_updater_ = std::make_unique<GcsServerAddressUpdater>(
+      options_.raylet_ip_address, options_.node_manager_port,
+      [this](std::string ip, int port) {
+        absl::MutexLock lock(&gcs_server_address_mutex_);
+        gcs_server_address_.first = ip;
+        gcs_server_address_.second = port;
+      });
+
+  // Initialize gcs client.
+  // As the synchronous and the asynchronous context of redis client is not used in this
+  // gcs client. We would not open connection for it by setting `enable_sync_conn` and
+  // `enable_async_conn` as false.
+  gcs::GcsClientOptions gcs_options = gcs::GcsClientOptions(
+      options_.gcs_options.server_ip_, options_.gcs_options.server_port_,
+      options_.gcs_options.password_,
+      /*enable_sync_conn=*/false, /*enable_async_conn=*/false,
+      /*enable_subscribe_conn=*/true);
+  gcs_client_ = std::make_shared<gcs::ServiceBasedGcsClient>(
+      gcs_options, [this](std::pair<std::string, int> *address) {
+        absl::MutexLock lock(&gcs_server_address_mutex_);
+        if (gcs_server_address_.second != 0) {
+          address->first = gcs_server_address_.first;
+          address->second = gcs_server_address_.second;
+          return true;
+        }
+        return false;
+      });
+
+  RAY_CHECK_OK(gcs_client_->Connect(io_service_));
```

Comment:
hmmm looks like now we are opening up multiple gcs connection from worker. AFAIK, python workers also connects to gcs within its python code. Now this adds separate connection from core worker to gcs server. I think this might be ok for now, but not sure what's the scalability limitation here. cc @rkooo567 @wuisawesome @scv119

Reply:
For the new thread to loop the gcs client in the core worker, I hope there is a better way. Currently, one core worker process may contains multiple core workers.

- core worker process (this PR add a new thread to loop a new io service in the core worker process)
  - core worker - io service
  - core worker - io service
  - ...
 
The stats is global in a process, it is shared by multiple core workers. The new thread does not cost too much, but it adds one more connection from core worker to the GCS.